File: 83.Remove_Duplicates_from_Sorted_List/83.Remove_Duplicates_from_Sorted_List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if(head === null || head.next === null) return head;
        let slow = head, fast = head.next;

        while(slow !== null && fast !== null ) {
            if(slow.val === fast.val) {
                slow.next = fast.next;
                fast = fast.next;
            } else {
                slow = fast;
                fast = fast.next;
            }
        }         
        return head;
};

File: 83.Remove_Duplicates_from_Sorted_List/83.Remove_Duplicates_from_Sorted_List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {

    public ListNode deleteDuplicates(ListNode head) {
        if(head == null || head.next == null) return head;
        ListNode slow = head, fast = head.next;

        while(slow != null && fast != null ) {
            if(slow.val == fast.val) {
                slow.next = fast.next;
                fast = fast.next;
            } else {
                slow = fast;
                fast = fast.next;
            }
        }         
        return head;
    }
}
